File: fesf/fesf-interface/src/main/java/ar/edu/fesf/wicket/application/MyApplication.java
package ar.edu.fesf.wicket.application;

import org.apache.wicket.Page;
import org.apache.wicket.authroles.authentication.AuthenticatedWebApplication;
import org.apache.wicket.authroles.authentication.AuthenticatedWebSession;
import org.apache.wicket.markup.html.WebPage;
import org.apache.wicket.spring.injection.annot.SpringBean;
import org.apache.wicket.spring.injection.annot.SpringComponentInjector;
import org.springframework.web.context.WebApplicationContext;
import org.springframework.web.context.support.WebApplicationContextUtils;

import ar.edu.fesf.services.SpringInitializedService;
import ar.edu.fesf.view.pages.home.Home;
import ar.edu.fesf.view.pages.librarian.LibrarianHome;
import ar.edu.fesf.view.pages.signin.SignIn;

public class MyApplication extends AuthenticatedWebApplication {

    private static final long serialVersionUID = 1L;

    private MounterURL aMounterURL;

    @SpringBean(name = "service.SpringInitializedService")
    private SpringInitializedService springInitializedService;

    public SpringInitializedService getSpringInitializedService() {
        return this.springInitializedService;
    }

    public void setSpringInitializedService(final SpringInitializedService springInitializedService) {
        this.springInitializedService = springInitializedService;
    }

    @Override
    public void init() {
        super.init();
        this.getMarkupSettings().setCompressWhitespace(true);
        this.getMarkupSettings().setStripWicketTags(true);

        this.setAMounterURL(new MounterURL(this));

        this.getComponentInstantiationListeners().add(new SpringComponentInjector(this, this.getSpringContext(), true));

        this.getSpringInitializedService().initialize();

        this.mountUrl("home", Home.class, "");
        this.mountUrl("librarian", LibrarianHome.class, "");
        this.mountUrl("login", SignIn.class, "");
    }

    private void mountUrl(final String mountPath, final Class<? extends WebPage> pageClass, final String... parameters) {
        this.getAMounterURL().mount(mountPath, pageClass, parameters);
    }

    @Override
    protected Class<? extends AuthenticatedWebSession> getWebSessionClass() {
        return SecuritySession.class;
    }

    @Override
    protected Class<? extends WebPage> getSignInPageClass() {
        return SignIn.class;
    }

    @Override
    public Class<? extends Page> getHomePage() {
        return Home.class;
    }

    /**
     * @return WebApplicationContext
     */
    public final WebApplicationContext getSpringContext() {
        return WebApplicationContextUtils.getWebApplicationContext(this.getServletContext());
    }

    public String getContextPath() {
        return this.getServletContext().getContextPath();
    }

    private void setAMounterURL(final MounterURL aMounter) {
        this.aMounterURL = aMounter;
    }

    private MounterURL getAMounterURL() {
        return this.aMounterURL;
    }

}

File: fesf/fesf-interface/src/main/scala/ar/edu/fesf/scala/view/application/MyApplication.scala
package ar.edu.fesf.scala.view.application
import org.apache.wicket.authroles.authentication.AuthenticatedWebApplication
import scala.reflect.BeanProperty
import org.apache.wicket.spring.injection.annot.SpringBean
import ar.edu.fesf.services.SpringInitializedService
import org.apache.wicket.spring.injection.annot.SpringComponentInjector
import org.apache.wicket.markup.html.WebPage
import ar.edu.fesf.wicket.application.MounterURL
import ar.edu.fesf.scala.view.pages.ScalaHome
import org.apache.wicket.authroles.authentication.AuthenticatedWebSession
import org.apache.wicket.Page
import org.springframework.web.context.WebApplicationContext
import org.springframework.web.context.support.WebApplicationContextUtils
import ar.edu.fesf.scala.view.pages.SignIn
import ar.edu.fesf.scala.view.pages.UpsPage
import org.apache.wicket.settings.IExceptionSettings
import org.springframework.security.core.context.SecurityContextHolder
import ar.edu.fesf.security.FakeAuthentication
import ar.edu.fesf.scala.view.pages.SignInLdap


@SerialVersionUID(1L)
class MyApplication(authenticationMode: AuthenticationMode) extends AuthenticatedWebApplication {

  @SpringBean
  @BeanProperty
  var springInitializedService: SpringInitializedService = _
  
  val mounterURL: MounterURL = new MounterURL(this)

  override def init() = {
    super.init()
    this.getMarkupSettings().setCompressWhitespace(true)
    this.getMarkupSettings().setStripWicketTags(true)

    this.getComponentInstantiationListeners().add(new SpringComponentInjector(this, this.getSpringContext(), true))

    this.getSpringInitializedService().initialize()

    this.getApplicationSettings().setInternalErrorPage(classOf[UpsPage])
    this.getExceptionSettings().setUnexpectedExceptionDisplay(IExceptionSettings.SHOW_INTERNAL_ERROR_PAGE)

    this.mountUrl("home", classOf[ScalaHome])
    this.mountUrl("login", this.loginPage())
  }
  
  private def mountUrl(mountPath: String, pageClass: Class[_ <: WebPage]) {
    this.mounterURL.mount(mountPath, pageClass, "");
  }

  override def getWebSessionClass(): Class[_ <: AuthenticatedWebSession] = this.securitySession()

  override def getSignInPageClass(): Class[_ <: WebPage] = this.loginPage()

  override def getHomePage(): Class[_ <: Page] = classOf[ScalaHome]

  def getSpringContext(): WebApplicationContext = WebApplicationContextUtils.getWebApplicationContext(this.getServletContext())

  def getContextPath(): String = this.getServletContext().getContextPath()
  
  def loginPage(): Class[_ <: WebPage] = this.authenticationMode.loginPage()
  
  def securitySession(): Class[_ <: AuthenticatedWebSession] = this.authenticationMode.securitySession()



}